x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearningUsageTransportAction.java

```diff
@@ -346,9 +362,36 @@ private static void initializeStats(Map<String, Long> emptyStatsMap) {
     }
 
     private static void updateStats(Map<String, Long> statsMap, Long value) {
-        statsMap.compute("sum", (k, v) -> v + value);
-        statsMap.compute("min", (k, v) -> Math.min(v, value));
-        statsMap.compute("max", (k, v) -> Math.max(v, value));
+        statsMap.computeIfPresent("sum", (k, v) -> v + value);
+        statsMap.computeIfPresent("min", (k, v) -> Math.min(v, value));
+        statsMap.computeIfPresent("max", (k, v) -> Math.max(v, value));
+    }
+
+    private void addDeploymentStats(GetDeploymentStatsAction.Response response, Map<String, Object> inferenceUsage) {
+        StatsAccumulator modelSizes = new StatsAccumulator();
+        StatsAccumulator timeStats = new StatsAccumulator();
+        StatsAccumulator nodeDistribution = new StatsAccumulator();
+        response.getStats().results().forEach(stats -> {
+            modelSizes.add(stats.getModelSize().getBytes());
+            stats.getNodeStats().forEach(nodeStats -> {
+                timeStats.add(nodeStats.getAvgInferenceTime().orElse(0.0) * nodeStats.getInferenceCount().orElse(0L));
```

Comment:
I wonder if the total time makes much sense. Especially when `model_threads` is > 1, it doesn't tell us much. On the other hand, it'd be useful to have stats on the `avg_inference_time` so maybe we can just collect stats for that?

Reply:
Sorry, I think I misspoke in my earlier comment. What I meant to say is that multiplying the avg_inference_time with the inference_count per node and then dividing that with the number of nodes won't give us a correct measurement. We'd need to divide with the sum of `inference_count`. So, if we want this stat we need to calculate it manually.

Reply:
🤦